### PR TITLE
Make more stuff adjustable

### PR DESCRIPTION
**Pull request by C0D3-M4513R**
Also updates to nixos 24.05

Used in: https://github.com/Col-E/Recaf/pull/805

### CODE REVIEW

File: buildGradleApplication/default.nix [hunk 1]
```diff
@@ -14,13 +13,15 @@
   version,
   src,
   meta,
+  jdk ? pkgs.jdk,
+  gradle ? pkgs.gradle,
```
Comment (raphiz) [inline buildGradleApplication/default.nix:17]: I guess that's reasonable in some cases. However, in most cases, overlays is probably the way to go.

File: buildGradleApplication/default.nix [hunk 2]
```diff
@@ -14,13 +13,15 @@
...
   buildInputs ? [],
   nativeBuildInputs ? [],
   dependencyFilter ? depSpec: true,
   repositories ? ["https://plugins.gradle.org/m2/" "https://repo1.maven.org/maven2/"],
   verificationFile ? "gradle/verification-metadata.xml",
   buildTask ? ":installDist",
-  installLocaltion ? "build/install/*/",
+  installLocation ? "build/install/*/",
```
Comment (raphiz) [inline buildGradleApplication/default.nix:24]: :see_no_evil:

**Review by raphiz — APPROVED** [review]
Awesome, thank you for your contribution 🤩

File: buildGradleApplication/default.nix (hunk 1 above)
Comment (C0D3-M4513R) [reply buildGradleApplication/default.nix:17]: I don't know why I would use overlays here tbh.
That change was able to work for me with 0 issues.

Comment (raphiz) [reply buildGradleApplication/default.nix:17]: sure, it works perfectly fine. Whenever possible, I prefer to have a single version of a JDK for - and by setting it explicitly in the overlay, it's used everywhere and not just for this package. But it's a matter of the use case I suppose 🤷.

**Pull request merged**